class-two-factor-core.php

```diff
@@ -785,12 +785,25 @@ public static function login_html( $user, $login_nonce, $redirect_to, $error_msg
 		<script>
 			(function() {
 				// Enforce numeric-only input for numeric inputmode elements.
-				var inputEl = document.querySelector( 'input.authcode[inputmode="numeric"]' );
+				const form = document.querySelector( '#loginform' ),
+					inputEl = document.querySelector( 'input.authcode[inputmode="numeric"]' ),
+					expectedLength = inputEl?.dataset.digits || 0;
+
 				if ( inputEl ) {
+					let spaceInserted = false;
 					inputEl.addEventListener(
 						'input',
 						function() {
-							this.value = this.value.replace( /[^0-9 ]/g, '' );
+							let value = this.value.replace( /[^0-9 ]/g, '' ).trimStart();
+
+							if ( ! spaceInserted && expectedLength && value.length === Math.floor( expectedLength / 2 ) ) {
+								value += ' ';
```

Comment:
This feels more complicated than it should be, but I don't see a good way to do it in CSS. Hopefully in the future we'll have [`nth-letter`](https://github.com/w3c/csswg-drafts/issues/3208) -- and  [`nth`-everything](https://css-tricks.com/a-call-for-nth-everything/), really 🤞🏻 

I did find [a workaround using `columns` and `word-wrap`](https://stackoverflow.com/a/24616034/450127), but I'm guessing it won't work inside an `<input>`. Maybe in a `<p contenteditable="true">`? That's also pretty hacky, though, and could cause other problems 🤷🏻‍♂️ 

The JS-based approach may be the most practical approach for now 👍🏻

Reply:
I was also hoping for a `nth-letter`, and saw a bunch of suggestions for using `<p contenteditable>` but was hoping not to go there.

Reply:
Yeah, I think the current approach is fine 👍🏻